build_runner/lib/src/generate/change_watcher.dart

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+import 'dart:async';
+
+import 'package:build/build.dart';
+import 'package:logging/logging.dart';
+import 'package:path/path.dart' as path;
+import 'package:watcher/watcher.dart';
+import 'directory_watcher_factory.dart';
+
+import '../package_graph/package_graph.dart';
+
+class AssetChange {
+  final AssetId id;
+  final ChangeType type;
+  AssetChange(this.id, this.type);
+}
+
+Future<Stream<AssetChange>> startFileWatchers(PackageGraph packageGraph,
+    Logger logger, DirectoryWatcherFactory directoryWatcherFactory) async {
+  var ctl = new StreamController<AssetChange>();
```

Comment:
When I saw this later on I didn't know what it was, lets spell out `controller`.